validator/src/commands/run/args/pub_sub_config.rs

```diff
@@ -11,20 +11,68 @@ use {
 
 static DEFAULT_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS: LazyLock<String> =
     LazyLock::new(|| PubSubConfig::default().max_active_subscriptions.to_string());
+static DEFAULT_TEST_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .max_active_subscriptions
+        .to_string()
+});
 
 static DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS: LazyLock<String> =
     LazyLock::new(|| PubSubConfig::default().queue_capacity_items.to_string());
+static DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .queue_capacity_items
+        .to_string()
+});
 
 static DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_BYTES: LazyLock<String> =
     LazyLock::new(|| PubSubConfig::default().queue_capacity_bytes.to_string());
+static DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_BYTES: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .queue_capacity_bytes
+        .to_string()
+});
 
 const DEFAULT_RPC_PUBSUB_WORKER_THREADS: &str = "4";
+static DEFAULT_TEST_RPC_PUBSUB_WORKER_THREADS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .worker_threads
+        .to_string()
+});
 
 #[cfg_attr(test, qualifiers(pub(crate)))]
 static DEFAULT_RPC_PUBSUB_NUM_NOTIFICATION_THREADS: LazyLock<String> =
     LazyLock::new(|| get_thread_count().to_string());
 
-pub(crate) fn args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
+pub(crate) fn args<'a, 'b, const TEST: bool>() -> Vec<Arg<'a, 'b>> {
+    let rpc_pubsub_notification_threads = Arg::with_name("rpc_pubsub_notification_threads")
+        .long("rpc-pubsub-notification-threads")
+        .takes_value(true)
+        .value_name("NUM_THREADS")
+        .validator(is_parsable::<usize>)
+        .help(
+            "The maximum number of threads that RPC PubSub will use for generating notifications. \
+             0 will disable RPC PubSub notifications",
+        );
+    let default_rpc_pubsub_max_active_subscriptions = if TEST {
+        &DEFAULT_TEST_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS
+    } else {
+        &DEFAULT_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS
+    }
+    .as_str();
+    let default_rpc_pubsub_queue_capacity_items = if TEST {
+        &DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS
+    } else {
+        &DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS
+    }
+    .as_str();
+    let default_rpc_pubsub_queue_capacity_bytes = if TEST {
+        &DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_BYTES
+    } else {
+        &DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_BYTES
+    }
+    .as_str();
```

Comment:
two nits:
- Are the `.as_str()`'s necessary ? We're referencing a `String`, but maybe they are necessary cuz of the LazyLock
- Can we just do one big if-else block at the top; make you sure you run `cargo fmt` as I'm not sure what the preferred formatting would be but something like this:
```rust
let (
    default_rpc_pubsub_max_active_subscriptions,
    default_rpc_pubsub_queue_capacity_items,
    default_rpc_pubsub_queue_capacity_bytes,
    rpc_pubsub_notification_threads
) = if test_validator {
    ...
} else {
    ...
};
```